src/storage/singletons/RatePlans/index.tsx

```
@@ -67,6 +61,117 @@ class RatePlans {
+          }>,
+        ) => {
+          const currentPriceHistory = data.data.prices;
+          runInAction(() => {
+            this.currentPriceHistory = currentPriceHistory;
+          });
+          successCallback && successCallback();
+          TableInfiniteScroll.setNewToken(data.data.next);
+        },
+      )
+      .catch(() => {
+        this.currentPriceHistory = [];
+      });
+  };
+
+  getMorePriceHistory = (
+    ratePlanId: string | number,
+    destinationId: string | number,
+    token: string,
+    setNewToken: (newToken: string) => void,
+    successCallback?: () => void,
+  ) => {
+    request({
+      route: `${configStore.config.draasInstance}/public/rate_plans/${ratePlanId}/destinations/${destinationId}/prices`,
+      payload: {
+        params: {
+          page_size: 10,
+          next: token,
+        },
+      },
+    })
+      .then(
+        (
+          data: AxiosResponse<{
+            prices: Array<DestinationPrice>;
+            next: string;
+          }>,
+        ) => {
+          const currentPriceHistory = data.data.prices;
+          runInAction(() => {
+            this.currentPriceHistory = toJS(this.currentPriceHistory).concat(
+              currentPriceHistory,
+            );
+          });
+          setNewToken(data.data.next);
+          successCallback && successCallback();
+        },
+      )
+      .catch(() => {
+        this.currentPriceHistory = [];
+      });
+  };
+
+  postNewDestination = (
+    ratePlanId: string | number,
+    payload: Pick<
+      RatePlanDetails,
+      "international" | "national" | "destinationId"
+    > & { startDate?: string },
+    successCallback?: () => void,
+  ) => {
+    request({
+      route: `${configStore.config.draasInstance}/public/rate_plans/${ratePlanId}/prices`,
+      loaderName: "@postNewDestination",
+      method: "post",
+      payload,
+    })
+      .then(() => {
+        successCallback && successCallback();
+      })
+      .catch(() => {});

```

empty catch? Can you remove or add errorNotification?